src/App.tsx

```diff
@@ -15,49 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  SORT_DEFAULT = '',
+  SORT_NAME = 'name',
+  SORT_LENGTH = 'lenght',
+}
+
+function getNewSortList(list:string[], sortValue:string, isReverse:boolean) {
+  const valueReset = [...list];
+
+  if (sortValue) {
+    valueReset.sort((val1, val2) => {
+      switch (sortValue) {
+        case SortBy.SORT_NAME:
+          return val1.localeCompare(val2);
+        case SortBy.SORT_LENGTH:
+          return val1.length - val2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    valueReset.reverse();
+  }
+
+  return valueReset;
+}
+
 export const App: React.FC = () => {
+  const [sortValue, setSortValue] = useState(SortBy.SORT_DEFAULT);
+  const [isReverse, setValueReverse] = useState(false);
+  const getNewList = getNewSortList(goodsFromServer, sortValue, isReverse);
```

Comment:
functions start with `verb`
```suggestion
  const newList = getNewSortList(goodsFromServer, sortValue, isReverse);
```